aws/resource_aws_appsync_datasource_test.go

```diff
@@ -384,6 +385,67 @@ func TestAccAwsAppsyncDatasource_Type_None(t *testing.T) {
 	})
 }
 
+func TestAccAwsAppsyncDatasource_Type_RelationalDatabase(t *testing.T) {
+	oldRegion := os.Getenv("AWS_DEFAULT_REGION")
+	os.Setenv("AWS_DEFAULT_REGION", "us-east-1")
+	defer os.Setenv("AWS_DEFAULT_REGION", oldRegion)
+	rName := fmt.Sprintf("tfacctest%d", acctest.RandInt())
+	resourceName := "aws_appsync_datasource.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSAppSync(t) },
+		Providers:    testAccProviders,
```

Comment:
You'll need to add a `ErrorCheck:   testAccErrorCheck(t, appsync.EndpointsID),` here and on line 429 to pass a relatively recent lint check performed by CI.